tests/public-api/examples/optimization/solver.py

```diff
@@ -0,0 +1,194 @@
+
+class OsparcSolver():
+    """
+    An oSparc solver
+    """
+    def __init__(self, solver_key: str, solver_version: str, cfg: osparc.Configuration):
+
+        self._solver_key: str = solver_key
+        self._solver_version: str = solver_version
+        self._cfg: osparc.Configuration = cfg
+
+        # APIs
+        try:
+            self._api_client = osparc.ApiClient(cfg)
+            self._users_api = osparc.UsersApi(self._api_client)
+            self._files_api = FilesApi(self._api_client)
+            self._solvers_api = SolversApi(self._api_client)
+            self._users_api.get_my_profile() # validate access
+        except osparc.exceptions.ApiException as e:
+            self._api_client.close()
+            expt = json.loads(e.body)
+            raise Exception('\n'.join(expt["errors"])) from None
+        
+        self._solver: Optional[Solver] = None
+
+        # Job dependent data
+        self._job: Optional[Job] = None
+        self._status: Optional[JobStatus] = None
+
+    def _generate_isolve_log(self) -> List[str]:
+        """
+        Unpacks the zip file containing iSolve logs and reads them in as a string
+        """
+        log_zip = Path(self._solvers_api.get_job_output_logfile(self._solver_key, self._solver_version, self._job.id))
+        log: List[str] = []
+        with TemporaryDirectory() as tmp_dir:
+            with zipfile.ZipFile(log_zip, 'r') as zip_ref:
+                zip_ref.extractall(tmp_dir)
+            for pth in Path(tmp_dir).iterdir():
+                if pth.is_file() and pth.name.endswith('.logs'):
+                    with open(pth, "r") as f:
+                        log.append(f.read())
+        os.remove(log_zip)
+        return log
+
+    def submit_job(self, input_file: Path):
+        """
+        submit job to solver
+        """
+        try:
+            # create objects
+            input: File = self._files_api.upload_file(file=input_file)
+            self._solver = self._solvers_api.get_solver_release(self._solver_key, self._solver_version)
+            self._job = self._solvers_api.create_job( self._solver.id, self._solver.version, JobInputs({"input_1": input}))
+
+            # solve
+            logging.info(f'Start solving job: {self._job.id}')
+
+            self._status = self._solvers_api.start_job(self._solver.id, self._solver.version, self._job.id)
+        except osparc.exceptions.ApiException as e:
+            self._api_client.close()
+            expt = json.loads(e.body)
+            raise Exception('\n'.join(expt["errors"])) from None
+                
+    def job_done(self) -> bool:
+        """
+        Check if a submitted job is done
+        """
+        try:
+            if self._job is None:
+                return True
+            self._status = self._solvers_api.inspect_job(self._solver_key, self._solver_version, self._job.id)
+            if self._status.stopped_at:
+                if self._status.state != 'SUCCESS':
```

Comment:
from looking at the `status` object it seems to me there are no other ways to do this. I suspect I could call another the other api function I use to get the iSolve log and that might provide some info about how the job exited. But I think maybe a better thing for me to do is to consider changing this so it can be read from the `status` object in a more "secure" way.